### PR TITLE
Apply "readability-redundant-access-specifiers" clang-tidy check

### PR DESCRIPTION
**Pull request by florian-kuebler**
Test: Compile

### CODE REVIEW

File: src/ConfigWidgets/include/ConfigWidgets/SymbolErrorDialog.h [hunk 1]
```diff
@@ -32,7 +32,6 @@ class SymbolErrorDialog : public QDialog {
   ~SymbolErrorDialog() override;
   [[nodiscard]] Result Exec();
 
- public slots:
```
Comment (dpallotti) [inline src/ConfigWidgets/include/ConfigWidgets/SymbolErrorDialog.h:35]: This removes Qt's `slots`. Is that fine?

File: src/ConfigWidgets/include/ConfigWidgets/SymbolLocationsDialog.h [hunk 2]
```diff
@@ -53,7 +53,6 @@ class SymbolLocationsDialog : public QDialog {
   // "module symbol file mapping" is saved and added to the list.
   ErrorMessageOr<void> TryAddSymbolFile(const std::filesystem::path& file_path);
 
- public slots:
```
Comment (dpallotti) [inline src/ConfigWidgets/include/ConfigWidgets/SymbolLocationsDialog.h:56]: Same

File: src/OrbitQt/include/OrbitQt/CallTreeWidget.h [hunk 3]
```diff
@@ -77,7 +77,6 @@ class CallTreeWidget : public QWidget {
   void OnSliderValueChanged(int value);
   void OnLeaveInspectionButtonClicked();
 
- private:
```
Comment (dpallotti) [inline src/OrbitQt/include/OrbitQt/CallTreeWidget.h:80]: These end up under `private slots`.

File: src/OrbitQt/include/OrbitQt/CaptureOptionsDialog.h [hunk 4]
```diff
@@ -120,7 +120,6 @@ class CaptureOptionsDialog : public QDialog {
   static constexpr uint16_t kMaxCopyRawStackSizeDefaultValue = 512;
   static constexpr uint16_t kThreadStateChangeMaxCopyRawStackSizeDefaultValue = 256;
 
- public slots:
```
Comment (dpallotti) [inline src/OrbitQt/include/OrbitQt/CaptureOptionsDialog.h:123]: This removes Qt's `slots`.

File: src/OrbitQt/include/OrbitQt/orbitsamplingreport.h [hunk 5]
```diff
@@ -45,7 +45,6 @@ class OrbitSamplingReport : public QWidget {
   void on_PreviousCallstackButton_clicked();
   void OnCurrentThreadTabChanged(int current_tab_index);
 
- private:
```
Comment (dpallotti) [inline src/OrbitQt/include/OrbitQt/orbitsamplingreport.h:48]: These end up under `private slots`.

File: src/OrbitQt/include/OrbitQt/orbittreeview.h [hunk 6]
```diff
@@ -64,7 +63,6 @@ class OrbitTreeView : public QTreeView {
   void OnDoubleClicked(QModelIndex index);
   void OnRowsSelected(std::vector<int>& rows);
 
- private:
```
Comment (dpallotti) [inline src/OrbitQt/include/OrbitQt/orbittreeview.h:67]: These end up under `private slots`.

File: src/OrbitQt/include/OrbitQt/orbittreeview.h [hunk 7]
```diff
@@ -52,7 +52,6 @@ class OrbitTreeView : public QTreeView {
   void SetIsInternalRefresh(bool status) { is_internal_refresh_ = status; }
   void SetIsMultiSelection(bool status) { is_multi_selection_ = status; }
 
- public slots:
```
Comment (dpallotti) [inline src/OrbitQt/include/OrbitQt/orbittreeview.h:55]: This removes Qt's `slots`.

File: src/SessionSetup/include/SessionSetup/ProcessLauncherWidget.h [hunk 8]
```diff
@@ -36,7 +36,6 @@ class ProcessLauncherWidget : public QWidget {
   void on_BrowseWorkingDirButton_clicked();
   void on_LaunchButton_clicked();
 
- private:
```
Comment (dpallotti) [inline src/SessionSetup/include/SessionSetup/ProcessLauncherWidget.h:39]: These end up under `private slots`.

**Review by dpallotti — COMMENTED** [review]
Please double check these mechanical changes. In this case, review the interaction with Qt's `slots`.

File: src/OrbitQt/include/OrbitQt/orbitsamplingreport.h (hunk 5 above)
Comment (florian-kuebler) [reply src/OrbitQt/include/OrbitQt/orbitsamplingreport.h:48]: Reverted the change and added a todo.

File: src/SessionSetup/include/SessionSetup/ProcessLauncherWidget.h (hunk 8 above)
Comment (florian-kuebler) [reply src/SessionSetup/include/SessionSetup/ProcessLauncherWidget.h:39]: Reverted the change and added a todo.

File: src/OrbitQt/include/OrbitQt/orbitsamplingreport.h [hunk 9]
```diff
@@ -45,7 +45,9 @@ class OrbitSamplingReport : public QWidget {
...
+  // TODO(https://github.com/google/orbit/issues/4589): Remove redundant "private" once slots is not
```
Comment (dpallotti) [inline src/OrbitQt/include/OrbitQt/orbitsamplingreport.h:48]: Should the `private slots:` above also have the relevant TODO like in the other files?

File: src/OrbitQt/include/OrbitQt/orbittreeview.h [hunk 10]
```diff
@@ -52,7 +52,9 @@ class OrbitTreeView : public QTreeView {
...
+  // TODO(https://github.com/google/orbit/issues/4589): Connect slots via code and not via UI files,
+  // and remove the "public slots" specifier
+ public slots:  // NOLINT(readability-redundant-access-specifiers)
   void columnResized(int column, int oldSize, int newSize);
 
  private slots:
```
Comment (dpallotti) [inline src/OrbitQt/include/OrbitQt/orbittreeview.h:60]: Same for this one.

File: src/SessionSetup/include/SessionSetup/ProcessLauncherWidget.h [hunk 11]
```diff
@@ -36,7 +36,9 @@ class ProcessLauncherWidget : public QWidget {
...
+  // TODO(https://github.com/google/orbit/issues/4589): Remove redundant "private" once slots is not
```
Comment (dpallotti) [inline src/SessionSetup/include/SessionSetup/ProcessLauncherWidget.h:39]: And same for the `private slots` above.

**Review by dpallotti — APPROVED** [review]
I left some comments on the consistency of where you added the TODOs: sometimes on the `private slots:`, sometime on the `private:`, sometimes on both.

**Review by beckerhe — APPROVED** [review]
I think this looks good now.

File: src/OrbitQt/include/OrbitQt/orbitsamplingreport.h (hunk 9 above)
Comment (florian-kuebler) [reply src/OrbitQt/include/OrbitQt/orbitsamplingreport.h:50]: Done.

File: src/OrbitQt/include/OrbitQt/orbittreeview.h (hunk 10 above)
Comment (florian-kuebler) [reply src/OrbitQt/include/OrbitQt/orbittreeview.h:62]: Done.

File: src/SessionSetup/include/SessionSetup/ProcessLauncherWidget.h (hunk 11 above)
Comment (florian-kuebler) [reply src/SessionSetup/include/SessionSetup/ProcessLauncherWidget.h:41]: Done.

**Pull request merged**